### ISSUE TITLE
Use `box3d_overal` for 2D non-axis aligned boxes

### ISSUE THREAD

**Issue by lioutasb** [facebookresearch/pytorch3d#1178]
Would it be possible to use [box3d_overal](https://pytorch3d.org/docs/iou3d) for 2D non-axis aligned boxes? Not many efficient CUDA implementations exist to support this operation and I was wondering if this function can be used instead. So far I wasn't able to use it since when I expand the 4 corners of the 2D box to 8 and set the `z` dimension to the same value for both boxes I get a `NaN` output.

Below is a minimal example:
```
import torch
from pytorch3d.ops import box3d_overlap

corner1 = torch.tensor([[[ 0.1162,  1.7993],
                         [-1.6162,  0.7993],
                         [-0.1162, -1.7993],
                         [ 1.6162, -0.7993]]], device='cuda:0')
corner2 = torch.tensor([[[ 3.8281,  1.0000],
                         [ 1.0000,  3.8281],
                         [-1.8281,  1.0000],
                         [ 1.0000, -1.8281]]], device='cuda:0')

corner1_3d = torch.nn.functional.pad(corner1, (0, 1))
corner1_3d = torch.cat([corner1_3d, corner1_3d], dim=1)

corner2_3d = torch.nn.functional.pad(corner2, (0, 1))
corner2_3d = torch.cat([corner2_3d, corner2_3d], dim=1)

intersection_vol, iou_3d = box3d_overlap(corner1_3d, corner2_3d)
print(intersection_vol, iou_3d)  # intersection_vol=0 and iou_3d=NaN but should be ~4.8198 and ~0.2805
```

**Comment by gkioxari** [comment]
I tried reproducing your code. There are some weird things going on with your input specifications.

1. `corner1_3d` and `corner2_3d` do not form 3D boxes. If you print `corner1_3d` you get the following which is not a 3D box. 

```python
tensor([[[ 0.1162,  1.7993,  0.0000],
         [-1.6162,  0.7993,  0.0000],
         [-0.1162, -1.7993,  0.0000],
         [ 1.6162, -0.7993,  0.0000],
         [ 0.1162,  1.7993,  0.0000],
         [-1.6162,  0.7993,  0.0000],
         [-0.1162, -1.7993,  0.0000],
         [ 1.6162, -0.7993,  0.0000]]], device='cuda:0')
```

2. The inputs to the `box3d_overlap` are boxes of shape `Nx8x3` and `Mx8x3`. 8 are the vertices of the cube and 3 is for the (x,y,z) coordinates. The ordering is also important, as detailed in the docstrings

https://github.com/facebookresearch/pytorch3d/blob/9f443ed26b906775a49bb69f2d7259ef48cfb6c9/pytorch3d/ops/iou_box3d.py#L123-L129

3.  Here is code that runs what I believe you wanted to run

```python
corner1 = torch.tensor([[[0.1162, 1.7993], [-1.6162, 0.7993], [-0.1162, -1.7993], [1.6162, -0.7993]]], device="cuda:0")
corner2 = torch.tensor([[[3.8281, 1.0000], [1.0000, 3.8281], [-1.8281, 1.0000], [1.0000, -1.8281]]], device="cuda:0")
corner1_3d = torch.nn.functional.pad(corner1, (0, 1))
corner1_3d = torch.cat([corner1_3d, corner1_3d], dim=1)

corner2_3d = torch.nn.functional.pad(corner2, (0, 1))
corner2_3d = torch.cat([corner2_3d, corner2_3d], dim=1)

corner1_3d[:, 4:, 2] = 1.0
corner2_3d[:, 4:, 2] = 1.0

intersection_vol, iou_3d = box3d_overlap(corner1_3d, corner2_3d)
```

which returns `iou_3d = 0.2806`
